### PR TITLE
Added more details to the Containerd CRI Plugin

### PR DESCRIPTION
**Pull request by raravena80**
- Changed the header a bit to reflect the new CRI Plugin name
- Also, added more details in terms of similarities and arch related to CRI-O

### CODE REVIEW

File: architecture.md [hunk 1]
```diff
@@ -644,9 +644,40 @@ a pod is **not** `Privileged` the runtime selection is done as follows:
 | Default CRI-O trust level: `untrusted` | Kata Containers                               | Kata Containers                              |  Kata Containers |
 
 
-### CRI-containerd
+### Containerd CRI Plugin (Formerly CRI-containerd)
```
Comment (jodh-intel) [inline architecture.md]: Alas, github markdown (unlike rST) cannot auto-generate the TOC, so please can you update the table of contents for this new section name?

File: architecture.md [hunk 2]
```diff
@@ -644,9 +644,40 @@ a pod is **not** `Privileged` the runtime selection is done as follows:
...
+
+The general guidelines for the Containerd CRI Plugin support is similar to the CRI-O support. You can run trusted workloads with a runtime like runC and then run an untrusted workload with Kata Containers. The parameters that can be modified in the containerd config to run Kata Containers along with another 'trusted' runtime are the following:
```
Comment (jodh-intel) [inline architecture.md]: Could you make that `runc` for consistency with the rest of the doc? There are a couple of instances where we aren't rendering that in a fixed font (which you could always fix in a separate commit if you want to ;)

File: architecture.md [hunk 3]
```diff
@@ -644,9 +644,40 @@ a pod is **not** `Privileged` the runtime selection is done as follows:
...
+
+```
```
Comment (jodh-intel) [inline architecture.md]: Could you specify the name of this config file?

File: architecture.md [hunk 4]
```diff
@@ -644,9 +644,40 @@ a pod is **not** `Privileged` the runtime selection is done as follows:
...
+      # runtime_type is the runtime type to use in containerd e.g. io.containerd.runtime.v1.linux
+      runtime_type = "io.containerd.runtime.v1.linux"
+
+      # runtime_engine is the name of the runtime engine used by containerd.
+      runtime_engine = ""
+
+      # runtime_root is the directory used by containerd for runtime state.
+      runtime_root = ""
+
+    # "plugins.cri.containerd.untrusted_workload_runtime" is a runtime to run untrusted workloads on it.
```
Comment (jodh-intel) [inline architecture.md]: It might be useful to show a completed example (with `kata-runtime` and `runc` specified)?

File: architecture.md (hunk 1 above)
Comment (raravena80) [reply architecture.md]: Thanks, @jodh-intel @jcvenegas should I amend the commit with the changes or create a new commit? would like to get a sense if there's a standard practice.

Comment (jcvenegas) [reply architecture.md]: We do amend and push again - to keep the semantic in the commit when modifications  correspond the the same change.

**Comment by jcvenegas** [comment]
hi @raravena80 ,  also pull approve require you "Signed-off-by" your commit.  Please amend and sign your commit :) .

**Comment by raravena80** [comment]
@jcvenegas @jodh-intel Amended changes. PTAL. Thx.

File: architecture.md [hunk 5]
```diff
@@ -202,7 +202,7 @@ The `kata-agent` makes use of [`libcontainer`](https://github.com/opencontainers
 to manage the lifecycle of the container. This way the `kata-agent` reuses most
 of the code used by [`runc`](https://github.com/opencontainers/runc).
 
-### Agend gRPC protocol
+### Agent gRPC protocol
```
Comment (jodh-intel) [inline architecture.md]: Good catch! :smile:

**Comment by jodh-intel** [comment]
Hi @raravena80 - thanks for updated. The commit message still need to be tweaked slightly to pass our automated checks. It's currently missing the one line summary (including the "subsystem"). Something like the following would make the checks happy as it includes one line summary with subsystem `docs`, a (single line) body, the "Fixes #XXX" comment and the SOB comment:

```
docs: Update arch doc for containerd

Added more details on the Containerd CRI plugin to the architecture document.

Fixes #77.
    
Signed-off-by: Ricardo Aravena <raravena@branch.io>
```

**Comment by jodh-intel** [comment]
Hi @raravena80 - the pullapprove.com check is complaining about the signed-off-by. I think it's because the email address in the commit != (one of?) the email addresses registered against your github account.

File: architecture.md [hunk 6]
```diff
@@ -644,13 +644,52 @@ a pod is **not** `Privileged` the runtime selection is done as follows:
 | Default CRI-O trust level: `untrusted` | Kata Containers                               | Kata Containers                              |  Kata Containers |
 
 
-### CRI-containerd
+### Containerd CRI Plugin (Formerly CRI-containerd)
+
+The general guidelines for the Containerd CRI Plugin support is similar to the CRI-O support. You can run trusted workloads with a runtime like `runc` and then run an untrusted workload with Kata Containers. The parameters that can be modified in the containerd config to run Kata Containers along with another 'trusted' runtime are the following:
```
Comment (klynnrif) [inline architecture.md]: ...The parameters that you can modify in the containerd...

Comment (raravena80) [reply architecture.md]: Added the change. Thx!

**Pull request merged**